content/articles/todolist-with-svelte-and-firebase/index.md

```diff
@@ -0,0 +1,1135 @@
+
+Paste this in the `home.svelte` file:
+
+```js
+<script>
+  import EditTodo from "../components/edit-todo.svelte";
+  import ViewTodos from "../components/view-todos.svelte";
+  import Header from "./../components/header.svelte";
+  import { UserStore } from "./../stores";
+  import page from "page";
+
+  $: if (!$UserStore) page.redirect("/login");
+</script>
+
+<section class="home">
+  <div class="wrapper">
+    <Header />
+    <EditTodo />
+    <ViewTodos />
+  </div>
+</section>
+
+<style>
+  section.home {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+  div.wrapper {
+    width: 100%;
+    max-width: 800px;
+  }
+</style>
+```
+
+This component is fairly short as it's purpose is to house all our other components. It also has a reactive statement that works similarly to the one in the `login` component. We simply check if there's no user currently logged in, and redirect to the `login` page if it holds true.
+
+The components imported in the `home` page have not yet been built so let's go ahead and build them. Create a `components` folder in the `src` folder and add `edit-todo.svelte`, `header.svelte`, and `view-todos.svelte` into it.
+
+Open the `edit-todo.svelte` file and paste this too:
+
+```js
+<script>
+    let name = "";
+    let priority = "1";
+  
+    const submitTodo = () => {
+      const newTodo = {
+        priority,
+        name,
+        completed: false,
+      };
+      console.log('Submit Todo', newTodo)
+    };
+  </script>
+  
+  <form>
+    <header>Add New Todo</header>
+  
+    <label for="name">
+      Todo Name
+      <input bind:value={name} id="name" type="text" />
+    </label>
+  
+    <label for="priority">
+      Todo Priority
+      <input
+        bind:value={priority}
+        id="priority"
+        min="1"
+        step="1"
+        max="5"
+        type="range"
+      />
+    </label>
+  
+    <button on:click|preventDefault={submitTodo}>Add Todo</button>
+  </form>
+  
+  <style lang="scss">
+    form {
+      background: #fff;
+      border-bottom-left-radius: 2rem;
+      border-bottom-right-radius: 2rem;
+      padding: 2rem 4rem;
+      header {
+        font-size: 2rem;
+        text-align: center;
+        font-weight: bold;
+        margin-bottom: 1rem;
+      }
+      label {
+        font-size: 1.4rem;
+        margin-bottom: 1rem;
+        display: block;
+        input {
+          display: block;
+          width: 100%;
+          margin-top: 0.5rem;
+          border-radius: 1rem;
+        }
+        &:nth-child(2) {
+          input {
+            padding: 0.5rem;
+          }
+        }
+      }
+      button {
+        padding: 1rem;
+        border: none;
+        background: #f2f9ff;
+        font-size: 1.4rem;
+        border-radius: 5px;
+        margin-top: 1rem;
+        border: solid 1px #ccc;
+        position: relative;
+        left: 50%;
+        transform: translateX(-50%);
+      }
+    }
+  </style>
```

Comment:
Try to avoid stylings in files, since it takes up too much space and reader might not have anything to learn there.

Instead, specify the URLs for that file.

Do this everywhere